gprofiler/client.py

```diff
@@ -122,8 +133,19 @@ def submit_profile(
                 "end_time": get_iso8601_format_time(end_time),
                 "hostname": self._hostname,
                 "profile": profile,
+                "cpu_avg": cpu_avg,
+                "mem_avg": mem_avg,
+                "spawn_time": get_iso8601_format_time_from_epoch_time(spawn_time),
             },
             timeout=self._upload_timeout,
             api_version="v2",
             params={"samples": str(total_samples), "version": __version__},
         )
+
+    def submit_metadata(self, metadata: Dict):
+        return self.post(
+            "agent-metadata",
+            metadata,
+            timeout=self._upload_timeout,
```

Comment:
I'd expect `post()` to pass `timeout=self._upload_timeout` itself. The purpose of that API is to hide common request details. Fix it?

Reply:
This endpoint is not in use, we get the metadata with each profiler request

Reply:
Removed in #134